bentoml/cli/deployment.py

```diff
@@ -250,20 +246,60 @@ def apply(
             if wait:
                 result_state = get_state_after_await_action_complete(
                     yaitai_service=yatai_service,
-                    name=deployment_name,
+                    name=name,
                     namespace=namespace,
+                    message='Creating deployment...',
+                )
+                result.deployment.state.CopyFrom(result_state.state)
+
+            _echo('Finished create deployment {}'.format(name), CLI_COLOR_SUCCESS)
+            display_deployment_info(result.deployment, output)
+
+    @deploy.command(help='Apply model service deployment from yaml file')
+    @click.option("--file", callback=parse_yaml_file_or_string_callback)
```

Comment:
```
@click.option("--file", "yaml_content", callback=...)
def apply(yaml_content, output, wait):
```

Reply:
use click's `type=click.File`, which also handle "string" input properly, most unix shell program refer "-" to reading from stdin.  also you can remove the "read from string" case in the callback function